Create Entry in CHANGELOG for 25.0 Release

lornazh opened the pull request:
### What does this Pull Request accomplish?

Add a list of added and changed features regarding STL 25.0 release in the change log.

### Why should this Pull Request be merged?

This information helps customers better understand what does the release do.

### What testing has been done?

N/A

Mattjet27 commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+    - `NationalInstruments.SemiconductorTestLibrary.25.0.0.nupkg` now includes PDB files for `NationalInstruments.SemiconductorTestLibrary.Abstractions.dll` and `NationalInstruments.SemiconductorTestLibrary.Extensions.dll`
+    - Source Link is now enabled on `NationalInstruments.SemiconductorTestLibrary.25.0.0.nupkg` which allows developers to debug into the source code of the Semiconductor Test Library for troubleshooting and understanding how it works.
+
+- ### Changed
```
@lornazh, we should add a new bullet (**Documentation & Examples**) and mention something along the lines as what I have shown below (note that you can omit the specific PRs, I just added them here for your context).

- **Documentation & Examples**
  - Various changes to examples and documentation to either fix, improve, or update them in accordance with user feedback and latest changes.
    - Fix examples for Constructing PinSiteData/SiteData with Site-agnostic Data (PR #194)
    - Fixed inline documentation for ConfigureOutputConnected extension method (#210).
     - Minor fix to example code shown in WritingTestCode user guide topic (PR #171)
    - Add missing constructors to provide additional documentation for certain classes (#175 & #211).
    - Updated the Sharing PinSiteData Between CodeModules user guide topic to reflect latest changes (#176 & #193)

lornazh replied:
Updated accordingly.

Mattjet27 reviewed (changes requested):
@lornazh, in addition to my inline comments and feedback, note that I found a few items that are missing from the CHANGELOG, can you you please capture these?
	• DAQ WriteDigital function throwing errors & DAQ measure of input tasks return wrong results (#156)
	•  DCPower.Measure.FetchMeasurement Extension Method Returns Voltage Twice Instead of Voltage and Current Measurements (#204)
	• Make VerifyTaskType public (#192)

lornazh commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+      - `TSMSessionManager.DAQmx(string[] pins, DAQmxTaskType expectedTaskType)`
+      - `TSMSessionManager.DAQmx(string[] pins, bool filterPins, DAQmxTaskType expectedTaskType)`
+    - New `Digital.LevelsAndTiming` extension methods added to configure and retrieve some time set related information
+      - `ConfigureTimeSetEdge( string timeSet, TimeSetEdge edge, double time)`
```
Fixed.

lornazh commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+  - **Instrument Abstraction**
+    - For DAQmx, now provides `PinSiteData<double> GetSampleClockRate()` extension method, and obsoletes `double[] GetSampleClockRates()` and `double GetSampleClockRateDistinct()`.
+  - **TestStandSteps**
+    - `SetupNIDigitalPatternInstruementation` now provides a new boolean input `applySourceWaveformData` to control whether to apply the data in waveform files to source waveforms.
```
Fixed.

lornazh commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+        - `SiteData<T> GetGlobalSiteData<T>(string dataId, bool filterForActiveSites = true)`
+        - `SetGlobalPinSiteData<T>(string dataId, PinSiteData<T> pinSiteData, bool overrideIfExisting = true)`
+        - `PinSiteData<T> GetGlobalPinSiteData<T>(string dataId, bool filterForActiveSites = true)`
+      - New `TryGetValue` method added to `SiteData<T>`
```
Fixed.

lornazh commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+  - **Data Abstraction**
+    - New `ISemiconductorModuleContext` extension methods added for sharing data between NI TestStand code modules
+      - `SetGlobalSiteData<T>(string dataId, SiteData<T> siteData, bool overrideIfExisting = true)`
+        - `SiteData<T> GetGlobalSiteData<T>(string dataId, bool filterForActiveSites = true)`
```
Fixed.

lornazh commented on CHANGELOG.md:
```diff
@@ -1,13 +1,57 @@
+      - `PinSiteData<DriveFormat> GetTimeSetDriveFormat(string timeSet)`
+  - **NuGet Package**
+    - `NationalInstruments.SemiconductorTestLibrary.25.0.0.nupkg` now includes PDB files for `NationalInstruments.SemiconductorTestLibrary.Abstractions.dll` and `NationalInstruments.SemiconductorTestLibrary.Extensions.dll`
+    - Source Link is now enabled on `NationalInstruments.SemiconductorTestLibrary.25.0.0.nupkg` which allows developers to debug into the source code of the Semiconductor Test Library for troubleshooting and understanding how it works.
```
Updated accordingly.

lornazh commented:
> @lornazh, in addition to my inline comments and feedback, note that I found a few items that are missing from the CHANGELOG, can you you please capture these? • DAQ WriteDigital function throwing errors & DAQ measure of input tasks return wrong results (#156) • DCPower.Measure.FetchMeasurement Extension Method Returns Voltage Twice Instead of Voltage and Current Measurements (#204) • Make VerifyTaskType public (#192)

@Mattjet27 the ones listed above are added to the Changed section.

Besides, I added one more
- Default evaluation type for default BurstPattern() method is Numeric Limit instead of Pass/Fail (#150)

Mattjet27 commented:
> > @lornazh, in addition to my inline comments and feedback, note that I found a few items that are missing from the CHANGELOG, can you you please capture these? • DAQ WriteDigital function throwing errors & DAQ measure of input tasks return wrong results (#156) • DCPower.Measure.FetchMeasurement Extension Method Returns Voltage Twice Instead of Voltage and Current Measurements (#204) • Make VerifyTaskType public (#192)
> 
> @Mattjet27 the ones listed above are added to the Changed section.
> 
> Besides, I added one more
> 
>     * [[BUG] Default evaluation type for default BurstPattern() method is Numeric Limit instead of Pass/Fail #150](https://github.com/ni/semi-test-library-dotnet/issues/150)

I purposefully did not want to include #150 since that issue is not directly tied to any of the code managed by this repo. The ini file that fix was implemented in is not managed by this project, rather it is an installed file part of STS Software itself. Please remove this from the CHANGELOG, but once you I will approve.

lornazh commented:
> > > @lornazh, in addition to my inline comments and feedback, note that I found a few items that are missing from the CHANGELOG, can you you please capture these? • DAQ WriteDigital function throwing errors & DAQ measure of input tasks return wrong results (#156) • DCPower.Measure.FetchMeasurement Extension Method Returns Voltage Twice Instead of Voltage and Current Measurements (#204) • Make VerifyTaskType public (#192)
> > 
> > 
> > @Mattjet27 the ones listed above are added to the Changed section.
> > Besides, I added one more
> > ```
> > * [[BUG] Default evaluation type for default BurstPattern() method is Numeric Limit instead of Pass/Fail #150](https://github.com/ni/semi-test-library-dotnet/issues/150)
> > ```
> 
> @lornazh, I purposefully did not want to include #150 since that issue is not directly tied to any of the code managed by this repo. Note that the ini file the fix for that issue was implemented in is not managed by this project, rather it is an installed file part of STS Software itself. Please remove this from the CHANGELOG, but once you I do will approve. Thanks!
> 
> Note
> 
> We may decided to raise a separate PR before the release to update this for any other important documentation changes we make between now and then that we want to capture in the CHANGELOG. That PR may need to be cherry-picked to the release branch if it raised after we branch.

Removed.
